File: growing-shop-auth-before/src/main/java/com/example/growingshopauth/auth/dto/RoleResponse.java
package com.example.growingshopauth.auth.dto;

import com.example.growingshopauth.auth.domain.Policy;
import com.example.growingshopauth.auth.domain.Role;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

public class RoleResponse {

    @NoArgsConstructor(access = AccessLevel.PROTECTED)
    @AllArgsConstructor
    @Builder
    @Getter
    public static class RoleRes {
        private Long id;
        private String name;
        private List<PoliciesRes> policies = new ArrayList<>();

        public static RoleRes from(Role role) {
            RoleRes res = new RoleRes();

            res.id = role.getId();
            res.name = role.getName();
            res.policies.addAll(
                    role.getPolicies().toResponse()
            );

            return res;
        }
    }

    @NoArgsConstructor(access = AccessLevel.PROTECTED)
    @AllArgsConstructor
    @Builder
    @Getter
    public static class PoliciesRes {
        private Long id;
        private String name;
        private String path;
        private String description;

        public static PoliciesRes from(Policy policy) {
            PoliciesRes res = new PoliciesRes();

            res.id = policy.getId();
            res.name = policy.getName();
            res.path = policy.getPath();
            res.description = policy.getDescription();

            return res;
        }
    }
}

File: growing-shop-auth/src/main/kotlin/com/example/growingshopauth/auth/dto/RoleResponse.kt
package com.example.growingshopauth.auth.dto

import com.example.domain.auth.Policy
import com.example.domain.auth.Role
import com.example.growingshopauth.auth.domain.toResponse

class RoleResponse {

    class RoleRes(
        val id: Long,
        val name: String,
        val policies: List<PoliciesRes> = listOf()
    ) {
        companion object {
            fun from(role: Role): RoleRes {
                return RoleRes(
                    role.id!!,
                    role.name,
                    role.policies.toResponse()
                )
            }
        }
    }

    class PoliciesRes(
        val id: Long,
        val name: String,
        val path: String,
        val description: String
    ) {
        companion object {
            fun from(policy: Policy): PoliciesRes {
                return PoliciesRes(
                    policy.id!!,
                    policy.name,
                    policy.path,
                    policy.description
                )
            }
        }
    }
}
